streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -688,6 +690,8 @@ class StreamingContext private[streaming] (
     } finally {
       // The state should always be Stopped after calling `stop()`, even if we haven't started yet
       state = STOPPED
+      // De-registering Streaming Metrics of the StreamingContext
+      env.metricsSystem.removeSource(streamingSource)
```

Comment:
I think should be done try, after calling scheduler.stop(). In the finally this will fail, if the SparkContext has been stopped and the env is invalid.